content/references/rippled-api/api-conventions/serialization.md

```diff
@@ -0,0 +1,285 @@
+
+
+### Amount Fields
+[Amount]: #amount-fields
+
+The "Amount" type is a special field type that represents an amount of currency, either XRP or an issued currency. This type consists of two sub-types:
+
+- **XRP**
+    XRP is serialized as a 64-bit unsigned integer (big-endian order), except that the most significant bit is always 0 to indicate that it's XRP, and the second-most-significant bit is `1` to indicate that it is positive. Since the maximum amount of XRP (10<sup>17</sup> drops) only requires 57 bits, you can calculate XRP serialized format by taking standard 64-bit unsigned integer and performing a bitwise-OR with `0x4000000000000000`.
+- **Issued Currencies**
+    Issued currencies consist of three segments in order:
+    1. 64 bits indicating the amount in the [internal currency format](currency-formats.html#issued-currency-math). The first bit is `1` to indicate that this is not XRP.
+    2. 160 bits indicating the [currency code](currency-formats.html#currency-codes). The standard API converts 3-character codes such as "USD" into 160-bit codes using the [standard currency code format](currency-formats.html#standard-currency-codes), but custom 160-bit codes are also possible.
+    3. 160 bits indicating the issuer's Account ID. (See also: [Account Address Encoding](accounts.html#address-encoding))
+
+You can tell which of the two sub-types it is based on the first bit: `0` for XRP; `1` for issued currency.
+
+The following diagram shows the serialization formats for both XRP amounts and issued currency amounts:
+
+![XRP amounts have a "not XRP" bit, a sign bit, and 62 bits of precision. Issued currency amounts consist of a "not XRP" bit, a sign bit, an exponent (8 bits), mantissa (54 bits), currency code (160 bits), and issuer (160 bits).](img/serialization-amount.png)
+
+
+### Array Fields
+[STArray]: #array-fields
+
+Some transaction fields, such as `SignerEntries` (in [SignerListSet transactions][]) and [`Memos`](transaction-common-fields.html#memos-field), are arrays of objects (called the "STArray" type).
+
+Arrays contain several [object fields](#object-fields) in their native binary format in a specific order. In JSON, each array member is a JSON "wrapper" object with a single field, which is the name of the member object field. The value of that field is the ("inner") object itself.
+
+In the binary format, each member of the array has a Field ID prefix (based on the single key of the wrapper object) and contents (comprising the inner object, [serialized as an object](#object-fields)). To mark the end of an array, append an item with a "Field ID" of `0xf1` (the type code for array with field code of 1) and no contents.
+
+The following example shows the serialization format for an array (the `SignerEntries` field):
+
```

Comment:
A personal observation about the format of the graphic. It is incredibly effective in conveying the format of the field! The only thing that hung me up for a few seconds was the box with SignerEntries field in it. At first glance, its width seemed to indicate the segment or length of the SignerEntries field as a sub-segment of the entire width of the larger box. It is clearly a title after just a little more examination. Not sure if it would help to change that format so that it isn't connected to the larger box or maybe so that it isn't in a box? Maybe having it float disconnected is worse. I didn't read it as such in the Amount Fields section - so maybe it is a non-issue.

Reply:
Changed all the images in this diagram to go that way. It also alleviated a little of the clutter of the diagrams.

Updated [DOC-1549](https://ripplelabs.atlassian.net/browse/DOC-1549) to update the currency format diagrams to match (since that affects #481, this PR, and whatever follow-up work DOC-1549 entails.)